app/mixins/components/trigger-build.js

```diff
@@ -0,0 +1,93 @@
+import Mixin from '@ember/object/mixin';
+import { task, timeout } from 'ember-concurrency';
+import { inject as service } from '@ember/service';
+import config from 'travis/config/environment';
+
+export default Mixin.create({
+  api: service(),
+  flashes: service(),
+  router: service(),
+
+  submitBuildRequest: task(function* () {
+    const data = yield this.createBuild.perform();
+
+    if (data) {
+      let requestId = data.request.id;
+      let { searchDebounceRate } = config.intervals;
+      yield timeout(searchDebounceRate);
+      yield this.showRequestStatus.perform(this.get('repo.id'), requestId);
+    }
+  }),
```

Comment:
I notice that the branch searching task has the `restartable` task modifier, would we want that here as well? I think it may be necessary for the debouncing to work as expected.

And similarly, do we want any task modifiers on the other tasks in this file? Here's a good summary about them http://ember-concurrency.com/docs/task-concurrency

Reply:
hmmm, i am not sure. i believe it should have the `drop` modifier ... once a the button has been pressed once, and the build request has been submitted, we want subsequent clicks to be ignored (which i believe also should happen by the way of disabling the button (which doesn't happen at the moment), ... but yeah, it might be good to also slap this modifier on the service here?

Reply:
i've changed it to `drop` subsequent clicks ... and it seems that's doing the right thing. even if i click as hard and as quickly as i can i'll only ever manage to create a single request to that api endpoint: https://github.com/travis-ci/travis-web/pull/2399/commits/9fd4d69e87f394f0777c54231d13ed3a0c9f0221

Reply:
Sounds good, thanks!